cmd/chart-repo/utils.go

```diff
@@ -415,3 +426,45 @@ func chartTarballURL(r repo, cv chartVersion) string {
 	}
 	return source
 }
+
+func initNetClient(rootCACert, additionalCACertDir string) (*http.Client, *http.Transport, error) {
+	caCertPool := x509.NewCertPool()
+	systemCACert, err := ioutil.ReadFile(rootCACert)
+	if err != nil {
+		log.Printf("Unable to read system CA certs: %v", err)
+	}
+	caCertPool.AppendCertsFromPEM(systemCACert)
+
+	files, err := ioutil.ReadDir(additionalCACertDir)
+	if err == nil && len(files) > 0 {
+		for _, fStat := range files {
+			fullPath := path.Join(additionalCACertDir, fStat.Name())
+			file, err := os.Open(fullPath)
+			if err != nil {
+				return nil, nil, fmt.Errorf("Unable to read CA cert file: %v", err)
+			}
+			fStat, err = file.Stat()
+			if err != nil {
+				return nil, nil, fmt.Errorf("Unable to read CA cert file: %v", err)
+			}
+			if !fStat.IsDir() {
+				caCert, err := ioutil.ReadFile(fullPath)
+				if err != nil {
+					return nil, nil, fmt.Errorf("Unable to read CA cert file: %v", err)
+				}
+				ok := caCertPool.AppendCertsFromPEM(caCert)
+				log.Printf("Adding CA cert from %s: %v", fullPath, ok)
+			}
+		}
+	}
+	// Return Transport for testing purposes
```

Comment:
can't we just access the transport using `client.Transport`?

Reply:
In any case I have changed the method and the test to avoid that and just check for errors.